pyproject.toml

```diff
@@ -73,16 +73,24 @@ isort.split-on-trailing-comma = false
 "tests/*" = ["D", "E501"] # TODO - remove E501
 
 [tool.pytest.ini_options]
-addopts = "--color=yes -p no:warnings --import-mode=importlib --durations=30"
+minversion = "6.0"
+addopts = ["-ra", "--showlocals", "--strict-markers", "--strict-config"]
```

Comment:
Please retain the previous options, too (although we can try without `-p no:warnings`. So please add

`--color=yes --import-mode=importlib --durations=30`

This might be why the tests are failing

Reply:
Sorry about that! I was going through the documentation and copied this in while initially running this against the repo review tool. I've added this back in now.

Reply:
The test failure looks like it's caused because of a package version issue. Could this be because I've set the `pytest` minimum version in the toml file to be 6.0 as per the repo review tool?

Reply:
Yeah I think it must be. It seems the `requests` library is now the wrong version:
https://stackoverflow.com/questions/73199376/requestsdependencywarning-urllib3-1-26-11-or-chardet-3-0-4-doesnt-match-a

Maybe try either 1) adding `requests` to `setup.cfg` as a testing dependency and/or 2) checking whether a newer version of `setuptools` (in pyproject.toml`) might address? Because from the test failure stacktrace, it seems that the requests packge is actually too new.

Reply:
```
 /usr/lib/python3/dist-packages/requests/__init__.py:69: in check_compatibility
    assert (3, 0, 2) <= (major, minor, patch) < (5, 0, 0)
E   AssertionError
        chardet_version = '5.2.0'
        major      = 5
        minor      = 2
        patch      = 0
        urllib3_version = ['1', '26', '5']
```